elide-core/src/main/java/com/yahoo/elide/parsers/state/RelationshipTerminalState.java

```diff
@@ -84,24 +84,26 @@ public RelationshipTerminalState(PersistentResource record, String relationshipN
 
     @Override
     public Supplier<Pair<Integer, JsonNode>> handlePatch(StateContext state) {
-        return handleRequest(state, this::patch);
+        return handleRequest(state, this::patch, "patch");
     }
 
     @Override
     public Supplier<Pair<Integer, JsonNode>> handlePost(StateContext state) {
-        return handleRequest(state, this::post);
+        return handleRequest(state, this::post, "post");
     }
 
     @Override
     public Supplier<Pair<Integer, JsonNode>> handleDelete(StateContext state) {
-        return handleRequest(state, this::delete);
+        return handleRequest(state, this::delete, "delete");
     }
 
-    private Supplier<Pair<Integer, JsonNode>> handleRequest(StateContext state,
-                                                           BiFunction<Data<Resource>, RequestScope, Boolean> handler) {
+    private Supplier<Pair<Integer, JsonNode>> handleRequest(
+            StateContext state,
+            BiFunction<Data<Resource>, RequestScope, Boolean> handler,
+            String requestType) {
         Data<Resource> data = state.getJsonApiDocument().getData();
         handler.apply(data, state.getRequestScope());
-        return () -> Pair.of(HttpStatus.SC_NO_CONTENT, null);
+        return constructResponse(record, state, requestType);
```

Comment:
This is [invalid](http://jsonapi.org/format/#crud-updating-relationship-responses-204)